near-sdk-macros/src/core_impl/contract_metadata/mod.rs

```diff
@@ -47,26 +53,37 @@ struct Standard {
 
 impl ContractMetadata {
     fn populate(mut self) -> Self {
-        if self.version.is_none() {
-            let version = std::env::var("CARGO_PKG_VERSION").unwrap_or(String::from(""));
-            if !version.is_empty() {
-                self.version = Some(version);
-            }
+        macro_rules! env_field {
+            ($field: expr, $env_result: expr) => {
+                if $field.is_none() {
+                    let field_val = $env_result.unwrap_or(String::from(""));
+                    if !field_val.is_empty() {
+                        $field = Some(field_val);
+                    }
+                }
+            };
         }
 
-        if self.link.is_none() {
-            let repo = std::env::var("CARGO_PKG_REPOSITORY").unwrap_or(String::from(""));
-            if !repo.is_empty() {
-                self.link = Some(repo);
-            }
-        }
+        env_field!(
+            self.link,
+            std::env::var("CARGO_NEAR_REPO_LINK_HINT").or(std::env::var("CARGO_PKG_REPOSITORY"))
+        );
+        env_field!(self.version, std::env::var("CARGO_PKG_VERSION"));
```

Comment:
Let's remove the macro, it takes more space and only makes it harder to read the code